src/main/java/net/rcarz/jiraclient/tempo/Team.java

```diff
@@ -0,0 +1,191 @@
+        JSON response = null;
+
+        try {
+            response = restclient.get(getRestUri());
+        } catch (Exception ex) {
+            throw new JiraException("Failed to retrieve all teams", ex);
+        }
+
+        if (!(response instanceof JSONArray)) {
+            throw new JiraException("JSON payload is malformed");
+        }
+        List<Team> result = new ArrayList<Team>();
+        Iterator it = ((JSONArray) response).iterator();
+        while (it != null && it.hasNext()) {
+            result.add(new Team((JSONObject) it.next()));
+        }
+
+        return result;
+    }
+
+    public static Team get(RestClient restClient, Integer id) throws JiraException{
+        JSON response = null;
+        if (id != null) {
+            try {
+                response = restClient.get(getRestUri() + id);
+            } catch (RestException rx) {
+                if (rx.getHttpStatusCode() == HttpStatus.SC_NOT_FOUND) {
+                    return null;
+                }
+            } catch (Exception ex) {
+                throw new JiraException("Failed to retrieve team by id: " + id, ex);
+            }
+        } else {
+            throw new JiraException("ID can't be null");
+        }
+        if (!(response instanceof JSONObject)) {
+            throw new JiraException("JSON payload is malformed");
+        }
+        JSONObject result = (JSONObject) response;
+        if (!result.containsKey(ID) || !(result.get(ID) instanceof Number)) {
+            return null;
+        }
+
+        return new Team(result);
+    }
+
+    public static Team create(RestClient restClient, String name, String summary, String lead) throws JiraException {
+        JSON result = null;
+
+        try {
+            result = restClient.post(getRestUri(), new Team(null, name, summary, lead).asJsonObject());
+        } catch (Exception ex) {
+            throw new JiraException("Failed to create team", ex);
+        }
+
+        if (!(result instanceof JSONObject) || !((JSONObject) result).containsKey(ID)
+                || !(((JSONObject) result).get(ID) instanceof Number)) {
+            throw new JiraException("Unexpected result on team creation: " + result.toString());
+        }
+
+        return new Team((JSONObject) result);
+    }
+
+    /**
+     * Delete tempo team by id
+     *
+     * @return false if team with provided id does not exist
+     */
+    public static boolean delete(RestClient restClient, Integer id) throws JiraException {
+        JSON result = new JSONObject();
+
+        if (id != null) {
+            try {
+                result = restClient.delete(getRestUri() + id.toString());
+            } catch (RestException rx) {
+                if (rx.getHttpStatusCode() == HttpStatus.SC_NOT_FOUND) {
+                    return false;
+                } else {
+                    throw new JiraException("Failed to delete team by id " + id, rx);
```

Comment:
probably it can be done easier simply "throw rx" - cause it will be caught 3 lines later and wrapped again